graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/RequiredTypeCollector.kt

```diff
@@ -44,16 +54,55 @@ class RequiredTypeCollector(
             object : NodeVisitorStub() {
                 val visitedTypes = mutableSetOf<String>()
 
+                override fun visitObjectTypeDefinition(
+                    node: ObjectTypeDefinition,
+                    context: TraverserContext<Node<*>>
+                ): TraversalControl {
+                    node.fieldDefinitions
+                        .flatMap { it.inputValueDefinitions }
+                        .forEach {
+                            it.type.findTypeDefinition(document)?.accept(context, this)
+                        }
+
+                    node.fieldDefinitions
+                        .filter { !visitedTypes.contains("${node.name}.${it.name}") }
+                        .forEach {
+                            visitedTypes.add("${node.name}.${it.name}")
+                            it.type.findTypeDefinition(document)?.accept(context, this)
+                        }
+                    return TraversalControl.CONTINUE
+                }
+
+                override fun visitUnionTypeDefinition(
+                    node: UnionTypeDefinition,
+                    context: TraverserContext<Node<*>>
+                ): TraversalControl {
+                    node.memberTypes.forEach {
+                        it.findTypeDefinition(document)?.accept(context, this)
+                    }
+                    return TraversalControl.CONTINUE
+                }
+
+                override fun visitTypeName(
+                    node: TypeName,
+                    context: TraverserContext<Node<*>>
+                ): TraversalControl {
+                    node.findTypeDefinition(document)?.accept(context, this)
```

Comment:
What does this line do?

Reply:
Is telling the tree-traversal to include _this node_.

Reply:
In the context of a _user_ selecting a _query_ this will add all _named types_ used by such selections to be included for _analysis_.

Reply:
Thanks!